src/umpire/strategy/DynamicPool.cpp

```diff
@@ -406,15 +422,10 @@ void DynamicPool::coalesce()
     const std::size_t released_bytes{releaseFreeBlocks()};
     // Deallocated and removed released_bytes from m_free_map
 
-    // If this removed anything from the map, re-allocate a single large chunk
+    // If this removed anything from the map, re-allocate a single large chunk and insert to free map
```

Comment:
The rounding here isn't needed because we always allocate rounded sizes, so `released_size` should always be rounded.